src/main/java/combat_system/CharacterStatusEffectFacade.java

```diff
@@ -0,0 +1,107 @@
+package combat_system;
+
+import characters.GameCharacter;
+
+import java.util.List;
+import java.util.Map;
+
+// TODO: add javadoc besides the ones i stole from george
+
+/**
+ * A facade class that provides a unified interface for Characters and StatusEffect interactions.
+ * All handling of status effects during combat occurs here.
+ */
+public class CharacterStatusEffectFacade {
+
+    /**
+     * Applies a status effect to a given target.
+     * If the target is already afflicted by the status effect,
+     * it will be reapplied if the remaining duration is lower than the new duration.
+     * @param target a reference to the target of the status effect
+     * @param effect the status effect to apply
+     * @param duration the duration of the status effect to apply
+     */
+    public void apply(GameCharacter target, StatusEffect effect, int duration) {
+        if (target.getStatusEffects().get(effect) == null ||
+                target.getStatusEffects().get(effect) < duration) {
+
+            target.setStatusEffect(effect, duration);
+
+            // Handling beginning of persistent subeffects
+            for (String subeffect : effect.getSubeffects()) {
+                // Splitting subeffect definition string
+                String[] parts = subeffect.split(" ");
+                String type = parts[0];
+                int argument = Integer.parseInt(parts[1]);
+
+                switch (type) {
+                    case "damage":
+                        changeAttack(target, argument);
+                        break;
+                    case "armor":
+                        changeDefense(target, argument);
+                        break;
+                }
+            }
+        }
+    }
+
+    public void process(List<GameCharacter> characters) {
+        for (GameCharacter character : characters) {
+            for (Map.Entry<StatusEffect, Integer> entry : character.getStatusEffects().entrySet()) {
+                StatusEffect effect = entry.getKey();
+                Integer duration = entry.getValue();
+
+                // Decrement status effects
+                character.setStatusEffect(effect, duration-1);
+
+                for (String subeffect : effect.getSubeffects()) {
+                    // Splitting subeffect definition string
+                    String[] parts = subeffect.split(" ");
```

Comment:
perhaps make the split of the subeffect a function? it looks like you use it multiple times.

Reply:
Good idea! In my latest commit, I have relegated the functionality of splitting the subeffect definition string to two separate methods.